aptos-move/move-examples/token_objects/knight/sources/knight.move

```diff
@@ -0,0 +1,275 @@
+        let property_mutator_ref = property_map::generate_mutator_ref(&constructor_ref);
+
+        // Transfers the token to the `soul_bound_to` address
+        let linear_transfer_ref = object::generate_linear_transfer_ref(&transfer_ref);
+        object::transfer_with_ref(linear_transfer_ref, receiver);
+
+        // Initializes the knight health point as 0
+        move_to(&object_signer, HealthPoint { value: 1 });
+
+        // Initialize the property map and the knight condition as Hungry
+        let properties = property_map::prepare_input(vector[], vector[], vector[]);
+        property_map::init(&constructor_ref, properties);
+        property_map::add_typed(
+            &property_mutator_ref,
+            string::utf8(CONDITION_PROPERTY_NAME),
+            string::utf8(CONDITION_HUNGRY),
+        );
+        property_map::add_typed(
+            &property_mutator_ref,
+            string::utf8(HEALTH_POINT_PROPERTY_NAME),
+            1,
+        );
+
+        // Publishes the KnightToken resource with the refs and the event handle for `HealthUpdateEvent`.
+        let knight_token = KnightToken {
+            mutator_ref,
+            property_mutator_ref,
+            health_update_events: object::new_event_handle(&object_signer),
+            base_uri
+        };
+        move_to(&object_signer, knight_token);
+    }
+
+    public entry fun feed_corn(from: &signer, to: Object<KnightToken>, amount: u64) acquires HealthPoint, KnightToken {
+        let corn_token = object::address_to_object<FoodToken>(food::corn_token_address());
+        feed_food(from, corn_token, to, amount);
+    }
+
+    public entry fun feed_meat(from: &signer, to: Object<KnightToken>, amount: u64) acquires HealthPoint, KnightToken {
+        let meat_token = object::address_to_object<FoodToken>(food::meat_token_address());
+        feed_food(from, meat_token, to, amount);
+    }
+
+    public entry fun feed_food(from: &signer, food_token: Object<FoodToken>, to: Object<KnightToken>, amount: u64) acquires HealthPoint, KnightToken {
+        let knight_token_address = object::object_address(&to);
+        food::burn_food(from, food_token, amount);
+
+        let restoration_amount = food::restoration_value(food_token) * amount;
+        let health_point = borrow_global_mut<HealthPoint>(object::object_address(&to));
+        let old_health_point = health_point.value;
+        let new_health_point = old_health_point + restoration_amount;
+        health_point.value = new_health_point;
+
+        let knight = borrow_global_mut<KnightToken>(knight_token_address);
+        // Gets `property_mutator_ref` to update the health point and condition in the property map.
+        let property_mutator_ref = &knight.property_mutator_ref;
+        // Updates the health point in the property map.
+        property_map::update_typed(property_mutator_ref, &string::utf8(HEALTH_POINT_PROPERTY_NAME), new_health_point);
```

Comment:
What's the purpose behind duplicating the health value into a property map and in the `HealthToken` resource on the knight token object? I think I am missing the original context of doing it this way, so was just wondering

Reply:
Ah, it's a temporary thing, in order to display the HealthPoint property in Petra. I will remove it once it's confirmed that Move resources are also displayed as properties in Petra, which @bowenyang007 has worked on.